spec/support/concerns/storage_provider_shared_examples.rb

```diff
@@ -0,0 +1,88 @@
+shared_context 'mocked StorageProvider Interface' do
+  let(:expected_chunk_max_exceeded) { false }
+  let(:expected_endpoint) { Faker::Internet.url }
+
+  before do
+    allow(mocked_storage_provider).to receive(:complete_chunked_upload)
+      .and_return(true)
+    allow(mocked_storage_provider).to receive(:max_chunked_upload_size)
+      .and_return(
+        mocked_storage_provider.chunk_max_number * mocked_storage_provider.chunk_max_size_bytes
+      )
+    allow(mocked_storage_provider).to receive(:chunk_max_exceeded?)
+      .and_return(expected_chunk_max_exceeded)
+    allow(mocked_storage_provider).to receive(:endpoint)
+      .and_return(expected_endpoint)
+    allow(mocked_storage_provider).to receive(:download_url) do |upload,filename=nil|
+      filename ||= upload.name
+      "#{expected_endpoint}/#{URI.encode(filename)}"
+    end
+    allow(mocked_storage_provider).to receive(:chunk_upload_url) do |chunk|
+      "#{expected_endpoint}/#{chunk.sub_path}"
+    end
+    allow(mocked_storage_provider).to receive(:suggested_minimum_chunk_size) do |upload|
+      (upload.size.to_f / mocked_storage_provider.chunk_max_number).ceil
+    end
+  end
+end
+
+shared_context 'mock all Uploads StorageProvider' do
+  # use this when uploads get created by other factories, and
+  # need their storage_provider mocked
+  include_context 'mocked StorageProvider'
+  include_context 'mocked StorageProvider Interface'
+
+  before do
+    allow_any_instance_of(Upload).to receive(:storage_provider)
+      .and_return(mocked_storage_provider)
+  end
+end
+
+shared_context 'mock Chunk StorageProvider' do |on: []|
+  let(:targets) {
+    if on.empty?
```

Comment:
Might be good to `expect(on).to be_an Array` before using it

Reply:
good idea